### PR TITLE
Added support for QML inside web-entities.

### PR DESCRIPTION
**Pull request by keeshii**
This PR allows to put QML files into web-entities:
- if the sourceUrl inside the web-entity ends with ".qml", then file will be parsed as QML,
- otherwise the content is treated as HTML.

The QML has similar capabilities as HTML and is not that resource hungry. Also the web-entities that are using QML do not count in regards the limit of 20 web-entities per domain.

I have rewritten the wizard from tutorial.json into QML. It may work as an example how the QML content can be written and communicated with the entity scripts: `interface/resources/serverless/Scripts/Wizard.qml`.

**Note: QML files must be whitelisted in the interface-app settings or they won't load. I've added "file" protocol to be whitelisted by default.**

### CODE REVIEW

File: libraries/entities-renderer/src/RenderableWebEntityItem.cpp [hunk 1]
```diff
@@ -66,6 +66,10 @@ WebEntityRenderer::ContentType WebEntityRenderer::getContentType(const QString&
 
     const QUrl url(urlString);
     auto scheme = url.scheme();
+    if (urlString.toLower().endsWith(".qml")) {
+        return ContentType::QmlContent;
+    }
+
```
Comment (ksuprynowicz) [inline libraries/entities-renderer/src/RenderableWebEntityItem.cpp:72]: Is this URL also checked against AUTHORIZED_EXTERNAL_QML_SOURCE?

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -3222,11 +3222,11 @@ void Application::initializeUi() {
 
             // END PULL SAFEURLS FROM INTERFACE.JSON Settings
 
-            if (AUTHORIZED_EXTERNAL_QML_SOURCE.isParentOf(url)) {
+            if (AUTHORIZED_EXTERNAL_QML_SOURCE.isParentOf(url) || url.scheme() == "file") {
```
[the remark this answers is not in the document]
Comment (keeshii) [reply interface/src/Application.cpp:3225]: About the webengine I am nearly sure that there is no possiblity to save a file on the disk without displaying "Save File Dialog".

From the JavaScript API there is no direct access to the file-system, but I know some resources are kept in the cache, for example `~/.local/share/Overte/Interface/data8/`. The files have no QML extension, so there is no possibility to execute them. Maybe there are some methods to upload a file I don't know.

Hm... I have added "file://" to the whitelist, beause I think it is useful (especially for development) and safe (probably). I am not a security nor QT/QML expert. I don't know the QML format well, so I don't know what is dangerous about them.

I can revert this change if you feel uncomfortable with it.

File: libraries/entities-renderer/src/RenderableWebEntityItem.cpp (hunk 1 above)
Comment (keeshii) [reply libraries/entities-renderer/src/RenderableWebEntityItem.cpp:72]: This is the code of the entities-renderer library.

The functions returns a value that represents the content displayed inside the web-entity. Based on that value, the "buildWebSurface" and "destroyWebSurface" methods are called. There is no need to create a WebEngine object if all we want is to display QML form.

Of course if the url to the QML file is not whitelisted, it won't be loaded inside the web-entity.

File: interface/src/Application.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (keeshii) [reply interface/src/Application.cpp]: Change reverted. The "file://" protocol is NOT whitelisted by default. User must add it manually in order to load QML files from disk.

**Review by ksuprynowicz — APPROVED** [review]
Everything looks fine :)

**Comment by ksuprynowicz** [comment]
The only thing we need to do before merging is QA testing - help will be very welcome. We need to test if QML opens inside web entity and if QMLs from file:// and non-whitelisted http:// URLs are rejected.

**Comment by ksuprynowicz** [comment]
I tested it, and WebEntity works fine but it is able to load files from local drive, for example:
file:/home/ksuprynowicz/overte/community-apps/applications/spectator-camera/SpectatorCamera.qml
loads the QML file.
Files from HTTP outside of whitelisted URLs get blocked as expected though.

**Comment by keeshii** [comment]
The qml whitelist should work similar as in current RC build. I will take a look, but I am quite possitive that the issue is not caused by my changes.

If the problem exists in the RC build. I would recommend fix it in a separate PR. It is an another/not related modification, that may cause other issues.

**Comment by ksuprynowicz** [comment]
> If the problem exists in the RC build. I would recommend fix it in a separate PR. It is an another/not related modification, that may cause other issues.

I agree. How can we check it in RC build?

**Comment by keeshii** [comment]
Try to copy-paste this in the console:
```
new OverlayWindow({ title: "Overlay Window", source: "file:///home/ksuprynowicz/overte/community-apps/applications/spectator-camera/SpectatorCamera.qml", width: 400, height: 300 });
```

**Comment by ksuprynowicz** [comment]
Thanks!
I'm merging it right now :)

**Pull request merged**